# make a custom cell chat database using kegg infomation.
# make from a specific kegg
# make from kegg 00001
# URI format https://www.genome.jp/kegg-bin/download_htext?htext=hsa00001&format=htext&filedir=kegg/brite/hsa
#' Get your kegg information.
#'
#' @param organs  Abbreviation of your species (hsa for human, mmu for mice, can be find in kegg website).
#' @param gtf A input gtf gene info generated by deal_gtf.
#'
#' @return kegg info.
#' @export
#'
#' @examples NA
make_kegg <- function(organs,gtf=NULL){
  kegURL <- c("https://www.genome.jp/kegg-bin/download_htext?htext=",organs,"00001&format=htext&filedir=kegg/brite/",organs)
  kegURL <- paste(kegURL,sep = "",collapse = "")
  print("Downloading keg file...")
  keg <- utils::read.csv(kegURL,header = F,sep = "\t")$V1
  print("Generating KO info...")
  ko <- c("NA",keg[stringr::str_starts(keg,"C")])
  ko <- cbind(substr(ko, start = 6, stop = 10),substr(ko, start = 12, stop = nchar(ko)))
  cname <- ko[1]
  k=1
  for (i in 1:length(keg)) {
    if (stringr::str_starts(keg[i],"C")){
      cname[i] <- ko[k]
      k = k + 1
    }
    else{
      cname[i] <- ko[k]
    }
  }
  names(keg) <- cname
  koGene <- keg[stringr::str_starts(keg,"D")]
  keglist <- stringr::str_extract(koGene, "\\d{2,9}")
  if (is.null(gtf)==F){
    keglist <- gtf$gene[match(keglist,gtf$db_xref)]
  }
  names(keglist) <- names(koGene)
  print("Almost done...")
  keglist <- split(unname(keglist),names(keglist))
  return(keglist)
}
